command/agent/config_test.go

```diff
@@ -211,6 +211,36 @@ func TestDecodeConfig(t *testing.T) {
 		t.Fatalf("bad: %#v", config)
 	}
 
+	// Advertise addresses for serflan
+	input = `{"advertise_addrs": {"serf_lan": "127.0.0.5:1234"}}`
+	config, err = DecodeConfig(bytes.NewReader([]byte(input)))
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if config.AdvertiseAddrs.SerfLanRaw != "127.0.0.5:1234" {
+		t.Fatalf("bad: %#v", config)
+	}
+
+	// Advertise addresses for serfwan
+	input = `{"advertise_addrs": {"serf_wan": "127.0.0.5:1234"}}`
+	config, err = DecodeConfig(bytes.NewReader([]byte(input)))
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if config.AdvertiseAddrs.SerfWanRaw != "127.0.0.5:1234" {
+		t.Fatalf("bad: %#v", config)
+	}
+
+	// Advertise addresses for rpc
+	input = `{"advertise_addrs": {"rpc": "127.0.0.5:1234"}}`
+	config, err = DecodeConfig(bytes.NewReader([]byte(input)))
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if config.AdvertiseAddrs.RPCRaw != "127.0.0.5:1234" {
+		t.Fatalf("bad: %#v", config)
+	}
```

Comment:
In these tests we should also test that the `*net.TCPAddr` is populated correctly.